src/main/java/ch/cern/eam/wshub/core/services/contractmanagement/impl/EquipmentReservationAdjustmentServiceImpl.java

```diff
@@ -0,0 +1,108 @@
+        MP7865_SyncCustomerRentalAdjustment_001 updateEntity = createUpdateEntity(context, equipmentReservationAdjustment);
+        MP7865_SyncCustomerRentalAdjustment_001_Result updateOperationResult = tools.performInforOperation(context, inforws::syncCustomerRentalAdjustmentOp, updateEntity);
+        return updateOperationResult.getResultData().getCUSTOMERRENTALADJUSTMENTID().getCUSTOMERRENTALADJUSTMENTPK();
+    }
+
+    @Override
+    public String deleteEquipmentReservationAdjustment(InforContext context, String number) throws InforException {
+        MP7866_DeleteCustomerRentalAdjustment_001 deleteEntity = createDeleteEntity(number);
+        MP7866_DeleteCustomerRentalAdjustment_001_Result deleteOperationResult = tools.performInforOperation(context, inforws::deleteCustomerRentalAdjustmentOp, deleteEntity);
+        return deleteOperationResult.getResultData().getCUSTOMERRENTALADJUSTMENTID().getCUSTOMERRENTALADJUSTMENTPK();
+    }
+
+    private MP7864_AddCustomerRentalAdjustment_001 createAddEntity(InforContext context, EquipmentReservationAdjustment equipmentReservationAdjustment) throws InforException {
+        MP7864_AddCustomerRentalAdjustment_001 addCustomerRentalAdjustment = new MP7864_AddCustomerRentalAdjustment_001();
+        CustomerRentalAdjustment customerRentalAdjustment = tools.getInforFieldTools().transformWSHubObject(createDefaultEquipmentReservationAdjustment(), equipmentReservationAdjustment, context);
+        addCustomerRentalAdjustment.setCustomerRentalAdjustment(customerRentalAdjustment);
+        return addCustomerRentalAdjustment;
+    }
+
+    private MP7863_GetCustomerRentalAdjustment_001 createGetEntity(String number) {
+        MP7863_GetCustomerRentalAdjustment_001 getCustomerRentalAdjustment = new MP7863_GetCustomerRentalAdjustment_001();
+        CUSTOMERRENTALADJUSTMENTID_Type idType = new CUSTOMERRENTALADJUSTMENTID_Type();
+        idType.setCUSTOMERRENTALADJUSTMENTPK(number);
+        getCustomerRentalAdjustment.setCUSTOMERRENTALADJUSTMENTID(idType);
+        return getCustomerRentalAdjustment;
+    }
+
+    private MP7865_SyncCustomerRentalAdjustment_001 createUpdateEntity(InforContext context, EquipmentReservationAdjustment equipmentReservationAdjustment) throws InforException {
+        MP7865_SyncCustomerRentalAdjustment_001 syncCustomerRentalAdjustment = new MP7865_SyncCustomerRentalAdjustment_001();
+        EquipmentReservationAdjustment prevReservation = readEquipmentReservationAdjustment(context, equipmentReservationAdjustment.getCode());
+        CustomerRentalAdjustment prevCustomerRentalAdjustment = tools.getInforFieldTools().transformWSHubObject(new CustomerRentalAdjustment(), prevReservation, context);
+        CustomerRentalAdjustment newCustomerRentalAdjustment = tools.getInforFieldTools().transformWSHubObject(prevCustomerRentalAdjustment, equipmentReservationAdjustment, context);
```

Comment:
We are calling `readEquipmentReservationAdjustment`, that is calling `transformInforObject` just to call again `transformWSHubObject` on the same object to get the operation result back again. Some refactoring needed here.